core/src/preferences/PreferencesBackend.cpp

```diff
@@ -104,20 +127,52 @@ namespace ledger {
             limitRaw[limitRaw.size() - 1] += 1;
             leveldb::Slice limit((const char *) limitRaw.data(), limitRaw.size());
             for (it->Seek(start); it->Valid(); it->Next()) {
-                fmt::print("Before test\n");
+                // TODO: decrypt here
                 if (it->key().compare(limit) > 0 || !f(it->key(), it->value())) {
                     break;
                 }
-                fmt::print("After all test\n");
             }
         }
 
         std::shared_ptr<Preferences> PreferencesBackend::getPreferences(const std::string &name) {
             return std::make_shared<Preferences>(*this, std::vector<uint8_t>(name.data(), name.data() + name.size()));
         }
 
-        PreferencesBackend::~PreferencesBackend() {
+        // Encrypt a preference change.
+        std::vector<uint8_t> PreferencesBackend::encrypt_preferences_change(
+            const std::shared_ptr<api::RandomNumberGenerator>& rng,
+            const std::string& password,
+            const PreferencesChange& change
+        ) {
+          // Create an AES256 cipher
+          const auto salt = std::string(std::begin(change.key), std::end(change.key));
```

Comment:
I’m not sure about that salt scheme at all. Here, `change.key` is the key in the key-value store.